examples/window/window_settings.rs

```diff
@@ -1,12 +1,23 @@
 //! Illustrates how to change window settings and shows how to affect
 //! the mouse pointer in various ways.
 
-use bevy::{prelude::*, window::PresentMode};
+use bevy::{
+    prelude::*,
+    window::{PresentMode, WindowIcon},
+};
 
 fn main() {
     App::new()
         .insert_resource(WindowDescriptor {
             title: "I am a window!".to_string(),
+            icon: Some(WindowIcon::new(
```

Comment:
Yeah this is *way* too low level for me to feel comfortable exposing directly to the user.

Are there reasonable choices for things to implement `From` for?

Reply:
Would `bevy::render::texture::Image` be a good fit?

Reply:
Ping @superdump for an informed opinion. That would be my natural first choice though.